### PR TITLE
Login

### PR DESCRIPTION
**Pull request by dsloan92**
Created the login page as well as logic to check whether or not a user is logged on.

### CODE REVIEW

**Comment by dsloan92** [comment]
Created login view and logic to handle whether or not a user is logged in.

File: controller/Controller.php [hunk 1]
```diff
@@ -202,6 +210,93 @@ function generateRandomString()
...
+                //login info was not valid set error message
+                $this->_f3->set('loginError', "Invalid Username and/or Password");
+            }
+
+        }
+        $view = new Template();
+        echo $view->render("views/login.html");
+    }
+
+    function logout()
```
Comment (kellerflint) [inline controller/Controller.php:277]: PHP Doc Header

File: controller/Controller.php [hunk 2]
```diff
@@ -202,6 +210,93 @@ function generateRandomString()
...
+                $this->_f3->reroute("/checklist/" . $_SESSION['user']->getUserID());
+            }
+        }
+    }
+
+    /**
+     * A function to check whether or not a user object has been set for the current session. If it is set the user
+     * can proceed to the page they were attempting to access. If it's not set, they are redirected to the login screen
+     */
+    private function isLoggedIn()
```
Comment (kellerflint) [inline controller/Controller.php:316]: @author

File: model/Database.php [hunk 3]
```diff
@@ -364,6 +364,29 @@ function getUserByEmail($user_email)
         return $statement->fetch(PDO::FETCH_ASSOC);
     }
 
+    /**
+     * Function to validate whether a user has provided valid credentials upon attempting to login
+     * @param string $username username input into login form
+     * @param string $password password input into login form
+     * @return mixed array containing userID of user logged in if valid, if not valid returns null
```
Comment (kellerflint) [inline model/Database.php]: idk what mixed means, id just delete it and use array

**Review by kellerflint — CHANGES_REQUESTED** [review]
Looks good overall, there are a few PHP docs things to add.

**Comment by dsloan92** [comment]
Have made the requested updated.

**Review by kellerflint — APPROVED** [review]

**Pull request merged**